physics/module_nst_water_prop.f90

```diff
@@ -12,6 +12,8 @@ module module_nst_water_prop
   public :: rhocoef,density,sw_rad,sw_rad_aw,sw_rad_sum,sw_rad_upper,sw_rad_upper_aw,sw_rad_skin,grv,solar_time_from_julian,compjd, &
             sw_ps_9b,sw_ps_9b_aw,get_dtzm_point,get_dtzm_2d
 
+  integer, parameter :: kp = kind_phys
```

Comment:
I know this probably predates your work, but why use two fields for the same thing and reference both throughout the code?
My suggestion(s) would be to modify the include statement:
- use machine, only : kind_phys
+ use machine, only : kp =>kind_phys
and use kp in the code. 
OR replace all instances of kp with kind_phys.

Reply:
Yes, I can definitely do that. I prefer ``use machine, only : kp =>kind_phys`` and then using as you say only kp. Mostly I was following what seemed typical style since I don't work on the ccpp side much.